### ISSUE TITLE
Speed ups possible for fmpz comparisons

### ISSUE THREAD

**Issue by albinahlback** [wbhart/flint2#1082]
One does not really need to call GMP for these comparisons, so we could save some extra cycles by not calling GMP.

**Comment by albinahlback** [comment]
I'm thinking of writing a reverse memcmp in assembly, so perhaps that will come in handy.

**Comment by wbhart** [comment]
The problem with assembly is it has to be written for each different processor. You aren't going to be able to beat GMP for that.

**Comment by albinahlback** [comment]
Yeah, I was only able to obtain ~1% speedup. Definitely not worth it.

**Comment by albinahlback** [comment]
But we can still call `mpn_cmp` instead of `mpz_cmp`, which should be inlined.
